<a href="https://colab.research.google.com/github/preethimulupoju/-technical-coding-assignment/blob/main/pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
"Import the necessary libraries"
import pandas as pd


Import the dataset from this(https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user).
Use sep= "|" while reading the data **bold text**
**bold text**

In [ ]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user'

Assign it to a variable called users and use the 'user_id' as index


In [ ]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user'


In [ ]:
users = pd.read_csv(url, sep='|', index_col='user_id')

In [ ]:
"See the first 10 and last 10 entries"
print("--------First 10 entries-------")
print(users.head(10).to_string())

print("--------Last 10 entries-------")
print(users.tail(10).to_string())



--------First 10 entries-------
         age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
2         53      F          other    94043
3         23      M         writer    32067
4         24      M     technician    43537
5         33      F          other    15213
6         42      M      executive    98101
7         57      M  administrator    91344
8         36      M  administrator    05201
9         29      M        student    01002
10        53      M         lawyer    90703
--------Last 10 entries-------
         age gender     occupation zip_code
user_id                                    
934       61      M       engineer    22902
935       42      M         doctor    66221
936       24      M          other    32789
937       48      M       educator    98072
938       38      F     technician    55038
939       26      F        student    33319
940       32      M  administrator    02215
941       20 

In [ ]:
# number of observations
print("Number of Observations : ", users.shape[0])

Number of Observations :  943


In [ ]:
# number of columns
print("Number of Columns : ", users.shape[1])

Number of Columns :  4


In [ ]:
# print column names
print("Column names: ")
print(users.columns)

Column names: 
Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')


In [ ]:
# print index information
print("Index information: ")
print(users.index)

Index information: 
Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)


<class 'pandas.core.frame.DataFrame'>
Int64Index: 943 entries, 1 to 943
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   age         943 non-null    int64 
 1   gender      943 non-null    object
 2   occupation  943 non-null    object
 3   zip_code    943 non-null    object
dtypes: int64(1), object(3)
memory usage: 36.8+ KB

In [ ]:
print(users.dtypes)


age            int64
gender        object
occupation    object
zip_code      object
dtype: object


In [ ]:
print(users['occupation'])


user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object


In [ ]:
num_occupations = users['occupation'].nunique()
print("Number of different occupations:", num_occupations)


Number of different occupations: 21


In [ ]:
most_frequent_occupation = users['occupation'].mode()[0]
print(f"Most frequent occupation: {most_frequent_occupation}, dtype: {type(most_frequent_occupation)}")



Most frequent occupation: student, dtype: <class 'str'>


In [ ]:
print(users.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 943 entries, 1 to 943
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   age         943 non-null    int64 
 1   gender      943 non-null    object
 2   occupation  943 non-null    object
 3   zip_code    943 non-null    object
dtypes: int64(1), object(3)
memory usage: 36.8+ KB
None


In [ ]:
print(users.describe(include='all'))


               age gender occupation zip_code
count   943.000000    943        943      943
unique         NaN      2         21      795
top            NaN      M    student    55414
freq           NaN    670        196        9
mean     34.051962    NaN        NaN      NaN
std      12.192740    NaN        NaN      NaN
min       7.000000    NaN        NaN      NaN
25%      25.000000    NaN        NaN      NaN
50%      31.000000    NaN        NaN      NaN
75%      43.000000    NaN        NaN      NaN
max      73.000000    NaN        NaN      NaN


In [ ]:
occupation_counts = users['occupation'].value_counts()
print(occupation_counts)


student          196
other            105
educator          95
administrator     79
engineer          67
programmer        66
librarian         51
writer            45
executive         32
scientist         31
artist            28
technician        27
marketing         26
entertainment     18
healthcare        16
retired           14
lawyer            12
salesman          12
none               9
homemaker          7
doctor             7
Name: occupation, dtype: int64


In [ ]:
mean_age = users['age'].mean()
print("Mean age of users:", mean_age)


Mean age of users: 34.05196182396607


In [ ]:
age_counts = users['age'].value_counts()
min_age = age_counts.idxmin()
print("Age with least occurrence:", min_age)


Age with least occurrence: 7
